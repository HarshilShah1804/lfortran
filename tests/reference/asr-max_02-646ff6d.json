{
    "basename": "asr-max_02-646ff6d",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/max_02.f90",
    "infile_hash": "c2180a170462597a12e62ea4c43eb4c64bd2a4759f66c8d393a12326",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-max_02-646ff6d.stdout",
    "stdout_hash": "c198e6cd162e623ee7e666b83aef8923c47e61d5a43f9d6c714cf255",
    "stderr": "asr-max_02-646ff6d.stderr",
    "stderr_hash": "97e360a47fa5edf6fba380dbc25904287f1994bc0091a7a77d421c5e",
    "returncode": 0
}