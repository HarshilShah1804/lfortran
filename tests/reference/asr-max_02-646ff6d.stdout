(TranslationUnit
    (SymbolTable
        1
        {
            max_02:
                (Program
                    (SymbolTable
                        2
                        {
                            y:
                                (Variable
                                    2
                                    y
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 8)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            z:
                                (Variable
                                    2
                                    z
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    max_02
                    []
                    [(Print
                        (StringFormat
                            ()
                            [(IntrinsicElementalFunction
                                Max
                                [(Cast
                                    (Var 2 y)
                                    IntegerToInteger
                                    (Integer 8)
                                    ()
                                )
                                (Cast
                                    (Var 2 z)
                                    IntegerToInteger
                                    (Integer 8)
                                    ()
                                )]
                                0
                                (Integer 8)
                                ()
                            )]
                            FormatFortran
                            (String -1 0 () PointerString)
                            ()
                        )
                    )
                    (Print
                        (StringFormat
                            ()
                            [(StringConstant
                                "Kind of max(y, z):"
                                (String 1 18 () PointerString)
                            )
                            (TypeInquiry
                                Kind
                                (Integer 8)
                                (IntrinsicElementalFunction
                                    Max
                                    [(Cast
                                        (Var 2 y)
                                        IntegerToInteger
                                        (Integer 8)
                                        ()
                                    )
                                    (Cast
                                        (Var 2 z)
                                        IntegerToInteger
                                        (Integer 8)
                                        ()
                                    )]
                                    0
                                    (Integer 8)
                                    ()
                                )
                                (Integer 4)
                                (IntegerConstant 8 (Integer 4) Decimal)
                            )]
                            FormatFortran
                            (String -1 0 () PointerString)
                            ()
                        )
                    )
                    (If
                        (IntegerCompare
                            (TypeInquiry
                                Kind
                                (Integer 8)
                                (IntrinsicElementalFunction
                                    Max
                                    [(Cast
                                        (Var 2 y)
                                        IntegerToInteger
                                        (Integer 8)
                                        ()
                                    )
                                    (Cast
                                        (Var 2 z)
                                        IntegerToInteger
                                        (Integer 8)
                                        ()
                                    )]
                                    0
                                    (Integer 8)
                                    ()
                                )
                                (Integer 4)
                                (IntegerConstant 8 (Integer 4) Decimal)
                            )
                            NotEq
                            (IntegerConstant 8 (Integer 4) Decimal)
                            (Logical 4)
                            (LogicalConstant
                                .false.
                                (Logical 4)
                            )
                        )
                        [(ErrorStop
                            (StringConstant
                                "Incorrect kind of max"
                                (String 1 21 () PointerString)
                            )
                        )]
                        []
                    )]
                )
        })
    []
)
